### DISCUSSION TITLE
L860-GL usb + Keenetic Viva

### DISCUSSION THREAD

**Discussion by Rusl2023** [bol-van/zapret#871] [Q&A]
Добрый день. Установить на внутренюю память роутера, при запуске скрипта выдает это:

/opt/zapret/init.d/sysv/zapret: line 19: syntax error: unexpected "then"
/ # /opt/zapret/init.d/sysv/zapret start
/opt/zapret/init.d/sysv/zapret: line 19: syntax error: unexpected "then"

видимо в конфиге че та напутал: 

#!/bin/sh
### BEGIN INIT INFO
# Provides:             zapret
# Required-Start:       $local_fs $network
# Required-Stop:        $local_fs $network
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
### END INIT INFO

SCRIPT=$(readlink -f "$0")
EXEDIR=$(dirname "$SCRIPT")
ZAPRET_BASE=$(readlink -f "$EXEDIR/../..")
. "$EXEDIR/functions"

NAME=zapret
DESC=anti-zapret

do_start()
{if lsmod | grep "xt_multiport " &> /dev/null ;  then
                echo "xt_multiport.ko is already loaded"
    else
        if insmod /lib/modules/$(uname -r)/xt_multiport.ko &> /dev/null; then
                        echo "iptable_raw.ko loaded"
        else
                        echo "Cannot find xt_multiport.ko kernel module, aborting"
                        exit 1
        fi
        fi

        if lsmod | grep "xt_connbytes " &> /dev/null ;  then
        echo "xt_connbytes.ko is already loaded"
    else
        if insmod /lib/modules/$(uname -r)/xt_connbytes.ko &> /dev/null; then
                        echo "xt_connbytes.ko loaded"
        else
                        echo "Cannot find xt_connbytes.ko kernel module, aborting"
                        exit 1
        fi
        fi

**Comment by bol-van** [comment]
Это не мой код, не принадлежит репозиторию, официально кинетик не поддерживается
К авторам все вопросы

**Comment by Fenrir-0xFF** [comment]
> /opt/zapret/init.d/sysv/zapret: line 19: syntax error: unexpected "then" 

Квадратные скобки поставь if [ "Условия" ]; then а выполняемые команды в ещё и в круглые перед которыми $ 
Пример if [ "$(lsmod | grep "xt_multiport " &> /dev/null)" ]; then
А вобще на днях [упрощённый скрипт](https://github.com/bol-van/zapret/issues/714) выложил какраз для железа с урезанными прошивками на котрых официальные скрипты zapret не тянут, главное чтобы бинарик nfqws запускался и перенаправления трафика в iptables хотябы на минимальном уровне работала.

**Comment by Rusl2023** [comment]
> Это не мой код, не принадлежит репозиторию, официально кинетик не поддерживается К авторам все вопросы

Да чуть модифицированный код. Вообщем решил вопрос вернувшись к предыдущей версии вашего кода, заработало. Грузит но не все, со стратегиями надо поработать. В 68 значений много, не разобрался пока что к чему. Спасибо за релизы, не стоишь на месте! 👍

**Comment by Rusl2023** [comment]
> > /opt/zapret/init.d/sysv/zapret: line 19: syntax error: unexpected "then"
> 
> Квадратные скобки поставь if [ "Условия" ]; then а выполняемые команды в ещё и в круглые перед которыми $ Пример if [ "$(lsmod | grep "xt_multiport " &> /dev/null)" ]; then А вобще на днях [упрощённый скрипт](https://github.com/bol-van/zapret/issues/714) выложил какраз для железа с урезанными прошивками на котрых официальные скрипты zapret не тянут, главное чтобы бинарик nfqws запускался и перенаправления трафика в iptables хотябы на минимальном уровне работала.

Спасибо бро, бум дальше вникать! 🙏
